File: src/main/java/action/junyoung/chapter04/Button.java
package action.junyoung.chapter04;

import action.junyoung.chapter04.part1.State;
import action.junyoung.chapter04.part1.View;
import org.jetbrains.annotations.NotNull;

public class Button implements View {

    @NotNull
    @Override
    public State getCurrentState() {
        return new ButtonState();
    }

    @Override
    public void restoreState(@NotNull State state) {}

    public class ButtonState implements State { }
}

File: src/main/kotlin/action/junyoung/chapter04/part1/Button.kt
package action.junyoung.chapter04.part1

class Button : Clickable, Focusable {
    override fun click() = println("I was clicked")
    override fun showOff() {
        super<Focusable>.showOff()
        super<Clickable>.showOff()
    }

    fun print() {
        println("I'm Button")
    }
}